roachpb/data.go

```diff
@@ -623,7 +623,7 @@ func (t *Transaction) Restart(userPriority, upgradePriority int32, timestamp Tim
 // others) for the transaction. If t.ID is empty, then the transaction is
 // copied from o.
 func (t *Transaction) Update(o *Transaction) {
-	if o == nil {
+	if o == nil || len(o.ID) == 0 {
```

Comment:
this is mostly an issue in a few tests which contain dummy errors carrying a non-nullable txn. This used to trigger the (then active) assertion below before I decided we had to remove it, so this should be fine to undo.